src/google/protobuf/descriptor_database.cc

```diff
@@ -628,7 +628,7 @@ bool EncodedDescriptorDatabase::DescriptorIndex::AddFile(const FileProto& file,
                                                          Value value) {
   // We push `value` into the array first. This is important because the AddXXX
   // functions below will expect it to be there.
-  all_values_.push_back({value.first, value.second});
+  all_values_.push_back({value.first, value.second, {}});
```

Comment:
@acozzette - should this be moved after the `ValidateSymbolName(file.package())` check and simply pass `file.package()`?  i.e. - why do we add the entry, then potentially return false and then updated what was added?

Reply:
I thought about that too but didn't want to also move it given that the mutation wouldn't happen in that case, but happy to if that's safe!

Reply:
Good question, I can't tell why it is done this way, but I hesitate to move the logic around given that the ordering does seem to be deliberate based on the comment.